querybook/server/lib/query_executor/executor_factory.py

```diff
@@ -33,16 +33,21 @@ def _get_executor_params_and_engine(query_execution_id, celery_task, session=Non
     if engine.deleted_at is not None:
         raise ArchivedQueryEngine("This query engine is disabled.")
 
+    proxy_user = user.username
+    executor_params = engine.get_engine_params()
+    if "proxy_user_id" in executor_params.keys():
+        try:
+            proxy_user = user.to_dict()[executor_params["proxy_user_id"]]
```

Comment:
you probably want some default value for executor_params["proxy_user_id"] if it is an empty string, say username

Reply:
Added in the above condition that executor_params["proxy_user_id"] shouldn't be null. If null, proxy_user would use the default user.username.

Reply:
i think we can simplify it to `if executor_params.get('proxy_user_id', '') != ''`